cmd/cronosd/cmd/fix-unlucky-tx.go

```diff
@@ -305,21 +312,27 @@ func newTxIndexer(config *tmcfg.Config, chainID string) (txindex.TxIndexer, erro
 	}
 }
 
-func findUnluckyTx(blockResult *tmstate.ABCIResponses) int {
+func (db *tmDB) findUnluckyTx(blockResult *tmstate.ABCIResponses, block *tmtypes.Block) (int, error) {
 	for txIndex, txResult := range blockResult.DeliverTxs {
 		if rpc.TxExceedsBlockGasLimit(txResult) {
-			return txIndex
+			tx := block.Txs[txIndex]
+			txHash := tx.Hash()
+			indexed, err := db.txIndexer.Get(txHash)
+			if err != nil {
+				return -1, err
+			}
+			if indexed != nil && (indexed.Result.Code == 0 || indexed.Height > block.Height) {
```

Comment:
```suggestion
			if indexed != nil && (indexed.Result.Code == 0 || indexed.Height > block.Height) {
```
still problematic after a second thought.
Conditions for skip:
- if `indexed != nil && indexed.Result.Code == 0`, tx was indexed with successful result, *DO SKIP*
- if `indexed != nil && indexed.Result.Code != 0 && txResult.Code == 0`, was indexed with failure result, and want to override with a successful result, "DO NOT SKIP"
- if `indexed != nil && indexed.Result.Code != 0 && txResult.Code != 0 && indexed.Height > block.Height`, was indexed with failure result, and want to override with a failure result, and was indexed at a later block, "DO SKIP"

It's a bit tricky, try to simplify the boolean logics here.

Reply:
So we should remove the indexed.Height > block.Height part, otherwise, the option 3 won’t hold, aka, although the tx is indexed by a later block, but it’s a failure result, and the current replayed result is a successful one, in this case we should not skip, right?